subgenerator/index.js

```diff
@@ -1,25 +1,25 @@
 'use strict';
 var path = require('path');
-var generators = require('yeoman-generator');
+var Generator = require('yeoman-generator');
 var superb = require('superb');
 
-module.exports = generators.Base.extend({
-  constructor: function () {
-    generators.Base.apply(this, arguments);
+module.exports = class extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
 
-    this.argument('namespace', {
+    this.option('name', {
```

Comment:
This can stay an argument. The issue is that namespace was a "reserved" name for yeoman-environment.